### PR TITLE
Support relative paths for task variable files

### PR DESCRIPTION
**Pull request by mkam**
Expands relative paths to full paths for the task variable files 
to support both relative and full paths.

Closes https://github.com/hashicorp/consul-terraform-sync/issues/279

### CODE REVIEW

File: client/terraform_cli.go [hunk 1]
```diff
@@ -77,11 +79,28 @@ func NewTerraformCLI(config *TerraformCLIConfig) (*TerraformCLI, error) {
 		log.Printf("[INFO] (client.terraformcli) persiting Terraform logs on disk: %s", logPath)
 	}
 
+	// Expand any relative paths for variable files to absolute paths
+	var varFiles []string
+	for _, vf := range config.VarFiles {
+		if !strings.HasPrefix(vf, "/") {
```
Comment (eikenb) [inline client/terraform_cli.go:85]: Consider using `path.IsAbs()` for this. A dedicated method makes the purpose more obvious.

Comment (eikenb) [reply client/terraform_cli.go:85]: Actually `filepath.IsAbs()` would probably be better. Already imported and better cross platform support.

File: client/terraform_cli.go [hunk 2]
```diff
@@ -77,11 +79,28 @@ func NewTerraformCLI(config *TerraformCLIConfig) (*TerraformCLI, error) {
...
+			wd, err := os.Getwd()
+			if err != nil {
+				log.Println("[ERR] (client.terraformcli) unable to retrieve current " +
+					"working directory to determine path to variable files")
+				log.Panic(err)
+			}
+			vfAbs := filepath.Join(wd, vf)
+			varFiles = append(varFiles, vfAbs)
+		} else {
+			varFiles = append(varFiles, vf)
```
Comment (eikenb) [inline client/terraform_cli.go:95]: `filepath.Join()` calls `filepath.Clean()` internally. Might be a good idea to Clean() these as well for consistency?

**Review by eikenb — APPROVED** [review]
LGTM.

**Review by findkim — APPROVED** [review]
Changes look good! I'm excited for consistency to support absolute and relative paths across the different configuration options 🎉

**Pull request merged**